### ISSUE TITLE
Fatal Exception: java.lang.NullPointerException: Activity must not be null

### ISSUE THREAD

**Issue by jplandry908** [michalchudziak/react-native-geolocation#217]
Hi @michalchudziak. I just released v3.0.3 within my app last night and starting to see some crashes in Crashlytics related to this module. I'm not able to reproduce the crash myself and it doesn't appear to be everyone, but the crash count is starting to climb. I'm also not seeing any similarities between the devices (device type, OS, etc).

I'm just curious if this may be related to the other issue since it is a similar 'Fatal Exception: java.lang.NullPointerException' error related to a NULL value. The way this module is implemented within the app has not changed in over a year, so the only change in this release was the module upgrade (upgraded from v2.1.0) and I have not seen this error before.
https://github.com/michalchudziak/react-native-geolocation/issues/206

Do you happen to know what may be causing this error or how I can help track down the root cause to provide you additional info? If I get any additional info, I will pass it along. Thank you!

Fatal Exception: java.lang.NullPointerException: Activity must not be null
       at com.google.android.gms.common.internal.Preconditions.checkNotNull(com.google.android.gms:play-services-basement@@18.1.0:2)
       at com.google.android.gms.common.api.internal.LifecycleActivity.<init>(com.google.android.gms:play-services-basement@@18.1.0:1)
       at com.google.android.gms.common.api.internal.LifecycleCallback.getFragment(com.google.android.gms:play-services-basement@@18.1.0:1)
       at com.google.android.gms.tasks.zzv.zza(com.google.android.gms:play-services-tasks@@18.0.1:1)
       at com.google.android.gms.tasks.zzw.addOnSuccessListener(com.google.android.gms:play-services-tasks@@18.0.1:3)
       at com.reactnativecommunity.geolocation.PlayServicesLocationManager.getCurrentLocationData(PlayServicesLocationManager.java:50)
       at com.reactnativecommunity.geolocation.GeolocationModule.lambda$getCurrentPosition$5$com-reactnativecommunity-geolocation-GeolocationModule(GeolocationModule.java:123)
       at com.reactnativecommunity.geolocation.GeolocationModule$$ExternalSyntheticLambda6.invoke(:8)
       at com.reactnativecommunity.geolocation.GeolocationModule.lambda$requestAuthorization$3(GeolocationModule.java:94)
       at com.reactnativecommunity.geolocation.GeolocationModule$$ExternalSyntheticLambda4.invoke(:11)
       at com.facebook.react.bridge.PromiseImpl.resolve(PromiseImpl.java:56)
       at com.facebook.react.modules.permissions.PermissionsModule.checkPermission(PermissionsModule.java:63)
       at com.reactnativecommunity.geolocation.GeolocationModule.requestAuthorization(GeolocationModule.java:98)
       at com.reactnativecommunity.geolocation.GeolocationModule.getCurrentPosition(GeolocationModule.java:123)
       at com.reactnativecommunity.geolocation.RNCGeolocationModule.getCurrentPosition(RNCGeolocationModule.java:41)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:188)
       at com.facebook.jni.NativeRunnable.run(NativeRunnable.java)
       at android.os.Handler.handleCallback(Handler.java:978)
       at android.os.Handler.dispatchMessage(Handler.java:104)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
       at android.os.Looper.loopOnce(Looper.java:238)
       at android.os.Looper.loop(Looper.java:357)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:228)
       at java.lang.Thread.run(Thread.java:1012)

**Comment by samzmann** [comment]
I'm also seeing similar reports in Sentry:
```
java.lang.NullPointerException: Activity must not be null
    at tg.s.k
    at sg.f.<init>
    at com.google.android.gms.common.api.internal.LifecycleCallback.c
    at bi.n0.l
    at bi.o0.f
    at com.reactnativecommunity.geolocation.p.b
    at com.reactnativecommunity.geolocation.GeolocationModule.getCurrentPosition
    at com.reactnativecommunity.geolocation.RNCGeolocationModule.getCurrentPosition
    at java.lang.reflect.Method.invoke(Method.java)
    at com.facebook.react.bridge.JavaMethodWrapper.invoke
    at com.facebook.react.bridge.JavaModuleWrapper.invoke
    at com.facebook.jni.NativeRunnable.run(SourceFile)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run
    at java.lang.Thread.run(Thread.java:920)
```

I'm not super comfy with Android coding, but from looking a little into the source code and reading some docs, I guess this is probably happening due to `mReactContext.getCurrentActivity()` possibly returning `null` in [PlayServicesLocationManager.java#L50](https://github.com/michalchudziak/react-native-geolocation/blob/master/android/src/main/java/com/reactnativecommunity/geolocation/PlayServicesLocationManager.java#L50)

The RN guide on building Android native modules shows how to check for Activity being null or not in this example: https://reactnative.dev/docs/native-modules-android.html#getting-activity-result-from-startactivityforresult
```java
@ReactMethod
public void pickImage(final Promise promise) {
  Activity currentActivity = getCurrentActivity();

  if (currentActivity == null) {
    promise.reject(E_ACTIVITY_DOES_NOT_EXIST, "Activity doesn't exist");
    return;
  }

  // run the actual logic
}
```

So I guess we could rewrite `getCurrentLocationData` like so
```java
 @Override
public void getCurrentLocationData(ReadableMap options, Callback success, Callback error) {
  AndroidLocationManager.LocationOptions locationOptions = AndroidLocationManager.LocationOptions.fromReactMap(options);

  Activity currentActivity = mReactContext.getCurrentActivity()

  if (currentActivity == null) {
    // return some kind of error by invoking the error callback, but don't crash the app
  }

  // now we can safely assume currentActivity is not null
  try {
    mFusedLocationClient.getLastLocation()
      .addOnSuccessListener(currentActivity, location -> {

     // rest would be the same

```

@michalchudziak What do you think?

**Comment by michalchudziak** [comment]
Hey! Thank you for reporting! @samzmann I think we can follow your suggestion - would you like to send a PR?

**Comment by samzmann** [comment]
@michalchudziak on it!

**Comment by samzmann** [comment]
@jplandry908 I patched 3.0.3 with the fix in PR https://github.com/michalchudziak/react-native-geolocation/pull/220 and not seeing any crashes so far. So I'd say you can consider this a working fix. If you want it before the next release of react-native-geolocation you can patch the package in your app.

@michalchudziak I think we can mark this issue as closed.
